src/frontend/src/lib/components/dapps/DappsCarousel.svelte

```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+	import { nonNullish } from '@dfinity/utils';
+	import Carousel from '$lib/components/carousel/Carousel.svelte';
+	import DappsCarouselSlide from '$lib/components/dapps/DappsCarouselSlide.svelte';
+	import {
+		dAppDescriptions,
+		type CarouselSlideOisyDappDescription
+	} from '$lib/types/oisyDappDescription';
+
+	export let styleClass: string | undefined = undefined;
+
+	const dappsCarouselSlides = dAppDescriptions.filter(({ carousel }) =>
+		nonNullish(carousel)
+	) as CarouselSlideOisyDappDescription[];
```

Comment:
do you need the casting? are you not taking elements of `dAppDescriptions`, that I think it's `OisyDappDescription` type?

Reply:
OisyDappDescription has `carousel` as optional. So, after filtering elements by this param, we can cast it to CarouselSlideOisyDappDescription where it's required.

Reply:
Thanks for the explanation!